SU2_CFD/src/output/COutput.cpp

```diff
@@ -654,19 +787,19 @@ void COutput::PrintConvergenceSummary(){
   ConvSummary.PrintHeader();
   for (unsigned short iField_Conv = 0; iField_Conv < convFields.size(); iField_Conv++){
     const string &convField = convFields[iField_Conv];
-    if (historyOutput_Map[convField].fieldType == HistoryFieldType::COEFFICIENT) {
+    if (historyOutput_Map.at(convField).fieldType == HistoryFieldType::COEFFICIENT) {
```

Comment:
out of interest: Why do you prefer .at over [] here? Is there something more than the out_of_bounds check and thrown exception?

Reply:
The [] operator for a map does not check whether that element exists. If it doesn't exist, it will be implicitly created, which can lead to strange behavior. Furthermore, you cannot use it in a const member function.

Reply:
Any performance difference? I know with .at() there is some overhead due to the extra checks, so for large data sets it may not be ideal. It shouldn't be noticeable for just a handful of checks though

Reply:
There shouldn't be any difference. Because the code for operator[] and at() is essentially the same. The only difference is that at() throws an exception and operator[] adds a new item in case the element cannot be found.

Reply:
For completeness on this [] vs .at() : .at() is a const operation on the data whilst [] isn't because [] creates the entry if it is not present in the map, as Tim states above. And now that we are adding more const's to the code and therefore const-correctness could become tricky one day -> .at() has your back.